### ISSUE TITLE
Doesn't animate properly on Android

### ISSUE THREAD

**Issue by BaseCase** [adorableio/react-native-animated-ellipsis#5]
which is like the whole point of this library, so that should probably work...

**Comment by relu-xz** [comment]
It's not working with Expo v24.0.0

**Comment by BaseCase** [comment]
Thanks for the note, @zorobabel. I'll make sure to test with that when I debug this.

**Comment by RzvnB** [comment]
Tested on Android device, and the dots are not animated.

**Comment by AylwSt01** [comment]
not animating on android... a bit disappointing! :(

**Comment by kyaroru** [comment]
hi all, i am able to fix it by altering the render method, because opacity does nothing on android.. so..
```
      if (o._value === 0) {
        return <Animated.Text key={key} style={{ color: 'transparent' }}> .</Animated.Text>;
      }
      return <Animated.Text key={key} style={{ opacity: o }}> .</Animated.Text>;
```

**Comment by BaseCase** [comment]
@kyaroru that's awesome! Thanks for figuring it out. I will try to find some time to incorporate your solution later this week.

**Comment by tangten** [comment]
@BaseCase 
```
// @flow

import React, { Component } from 'react'
import { View, Animated, StyleSheet } from 'react-native'
import { colors } from 'constants/index'

type Props = {
  numberOfDots: number,
  animationDelay: number,
  minOpacity: number,
  style: any
}

export default class AnimatedEllipsis extends Component<Props> {
  static propTypes = {}

  static defaultProps = {
    numberOfDots: 3,
    animationDelay: 300,
    minOpacity: 0,
    style: {
      color: colors.grey,
      fontSize: 40
    }
  }

  animationState: any

  constructor (props: Props) {
    super(props)

    this.animationState = {
      dotOpacities: this.initializeDots(),
      targetOpacity: 1,
      shouldAnimate: true
    }
  }

  initializeDots = () => {
    let opacities = []

    for (let i = 0; i < this.props.numberOfDots; i++) {
      let dot = new Animated.Value(this.props.minOpacity)
      opacities.push(dot)
    }

    return opacities
  }

  componentDidMount () {
    this.animateDots(0)
  }

  componentWillUnmount () {
    this.animationState.shouldAnimate = false
  }

  animateDots = (whichDot: number) => {
    if (!this.animationState.shouldAnimate) return

    if (whichDot >= this.animationState.dotOpacities.length) {
      whichDot = 0
      let min = this.props.minOpacity
      this.animationState.targetOpacity = this.animationState.targetOpacity === min ? 1 : min
    }

    let nextDot = whichDot + 1

    Animated.timing(this.animationState.dotOpacities[whichDot], {
      toValue: this.animationState.targetOpacity,
      duration: this.props.animationDelay
    }).start(this.animateDots.bind(this, nextDot))
  }

  render () {
    let dots = this.animationState.dotOpacities.map((o, index) => {
      return (
        <Animated.Text key={index} style={[this.props.style, { opacity: o }]}>
          {' '}
          .
        </Animated.Text>
      )
    })

    return <View style={styles.container}>{dots}</View>
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row'
  }
})
```

The real problem is `Animated.Text` can not work inside nested `Text` component. For my solution just replace `Text` by `View` and it works correctly now

**Comment by BaseCase** [comment]
Thanks again to @tangnv and @mcavaliere for the help on this! Just merged and published version `2.0.0`. I bumped the major version because a few things about the styling work different with this new, better structure.
